src/dftbp/common/memman.F90

```diff
@@ -5,14 +5,64 @@
 !  See the LICENSE file for terms of usage and distribution.                                       !
 !--------------------------------------------------------------------------------------------------!
 
+#:include 'common.fypp'
+
 !> Contains various constants for memory management
 module dftbp_common_memman
   use dftbp_common_accuracy, only : dp
+  use dftbp_io_message, only : error, warning
+  use, intrinsic :: iso_c_binding, only : c_ptr, c_null_ptr, c_f_pointer,&
+      & c_int, c_intptr_t, c_size_t
 
   implicit none
 
   private
-  public :: incrmntOfArray
+  public :: incrmntOfArray, TAlignedArray
+
+  !> Holds an allocated array being aligned in memory
+  type TAlignedArray
+
+    !> Actual array data
+    real(dp), pointer :: array(:)
+
+    !> Number of array elements
+    integer :: size = 0
+
+    !> Chosen alignment in bytes (default: 64 bytes)
+    integer :: alignment = 64
+
+    !> Whether the array is allocated
+    logical :: isAllocated = .false.
+
+    !> Pointer to the allocated memory
+    type(c_ptr) :: memoryPointer = c_null_ptr
+
+  contains
+
+    procedure :: allocateAligned => TAlignedArray_allocateAligned
```

Comment:
It is a matter of taste, and we can discuss, but I'd just call this `%allocate()`  instead of `%allocateAligned()`, as you can not allocate this array other than aligned.